### PR TITLE
update docs with X-Druid-SQL-Query-Id

### PR DESCRIPTION
**Pull request by vtlim**
This PR documents the "X-Druid-SQL-Query-Id" header.

<hr>

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: docs/querying/sql.md [hunk 1]
```diff
@@ -871,13 +871,13 @@ Submit your query as the value of a "query" field in the JSON object within the
...
 
-##### Request
+##### Request body
       
 |Property|Description|Default|
 |--------|----|-----------|
 |`query`|SQL query string.| none (required)|
 |`resultFormat`|Format of query results. See [Responses](#responses) for details.|`"object"`|
-|`header`|Whether or not to include a header. See [Responses] for details.|`false`|
+|`header`|Whether or not to include a header. See [Responses](#responses) for details.|`false`|
```
Comment (techdocsmith) [inline docs/querying/sql.md:880]: Maybe clarify further that this means header row for the result set, not having to do with the HTTP header.

File: docs/querying/sql.md [hunk 2]
```diff
@@ -952,6 +952,8 @@ You can additionally request a header by setting "header" to true in your reques
 In this case, the first result returned will be a header. For the `csv`, `array`, and `arrayLines` formats, the header
 will be a list of column names. For the `object` and `objectLines` formats, the header will be an object where the
 keys are column names, and the values are null.
+If you specified `sqlQueryId` in the [connection context parameters](#connection-context),
+the SQL query identifier will be returned in the header labeled `X-Druid-SQL-Query-Id`.
```
Comment (techdocsmith) [inline docs/querying/sql.md:956]:
```suggestion
Druid returns the SQL query identifier in the `X-Druid-SQL-Query-Id` header.
```
nit: prefer active

**Review by techdocsmith — CHANGES_REQUESTED** [review]

File: docs/querying/sql.md (hunk 2 above)
Comment (maytasm) [reply docs/querying/sql.md:956]: If you don't specified `sqlQueryId` in the connection context parameters, a sqlQueryId will be generated by Druid and return in the response's `X-Druid-SQL-Query-Id` header

File: docs/querying/sql.md [hunk 3]
```diff
@@ -877,7 +877,7 @@ Submit your query as the value of a "query" field in the JSON object within the
 |--------|----|-----------|
 |`query`|SQL query string.| none (required)|
 |`resultFormat`|Format of query results. See [Responses](#responses) for details.|`"object"`|
-|`header`|Whether or not to include a header. See [Responses](#responses) for details.|`false`|
+|`header`|Whether or not to include a header for the query response. See [Responses](#responses) for details.|`false`|
```
Comment (maytasm) [inline docs/querying/sql.md:880]: wait. I think there is some confusion here. The `header` here is talking about the column names of each column in the result. This is different for the header of the HTTP response.

**Review by maytasm — CHANGES_REQUESTED** [review]
Please see comment

File: docs/querying/sql.md (hunk 3 above)
Comment (vtlim) [reply docs/querying/sql.md:880]: @maytasm Yeah, that's what I was trying to differentiate but not very well. Maybe "query result" would be less confusing?

File: docs/querying/sql.md [hunk 4]
```diff
@@ -871,13 +871,13 @@ Submit your query as the value of a "query" field in the JSON object within the
...
 
-##### Request
+##### Request body
       
 |Property|Description|Default|
 |--------|----|-----------|
 |`query`|SQL query string.| none (required)|
 |`resultFormat`|Format of query results. See [Responses](#responses) for details.|`"object"`|
-|`header`|Whether or not to include a header. See [Responses] for details.|`false`|
+|`header`|Whether or not to include a header for the query result. See [Responses](#responses) for details.|`false`|
```
Comment (techdocsmith) [inline docs/querying/sql.md:880]:
```suggestion
|`header`|Whether or not to include a header row for the query result. See [Responses](#responses) for details.|`false`|
```
I think saying header row helps with the differentiation.

File: docs/querying/sql.md [hunk 5]
```diff
@@ -949,10 +949,14 @@ You can additionally request a header by setting "header" to true in your reques
 }
 ```
 
-In this case, the first result returned will be a header. For the `csv`, `array`, and `arrayLines` formats, the header
+In this case, the first result of the response body returned will be a header. For the `csv`, `array`, and `arrayLines` formats, the header
```
Comment (techdocsmith) [inline docs/querying/sql.md:952]:
```suggestion
In this case, the first result of the response body is the header row. For the `csv`, `array`, and `arrayLines` formats, the header
```

**Review by techdocsmith — APPROVED** [review]
Minor clarifications/ style

**Review by maytasm — APPROVED** [review]

**Pull request merged**